### ISSUE TITLE
How to force generate @Nullable/@Nonnull for getters/setters?

### ISSUE THREAD

**Issue by marcelmatula** [joelittlejohn/jsonschema2pojo#1512]

**Comment by unkish** [comment]
Hi

I'm afraid it's currently not supported, there is a similar request #897 for `jsr303`

**Comment by recombinationgroup** [comment]
I just want to emphasize that this feature would be important for improving interoperability of the generated sources with Kotlin. The Kotlin compiler only properly detects the POJO properties as nullable if the getters are annotated with @Nullable as well. Otherwise, the property will be treated as Non-Nullable at compile time, which shadows possible NPE sources.

**Comment by ctrimble** [comment]
@marcelmatula you could do this with a custom annotator.

Extend the AbstractAnnotator and override the propertySetter method.  In that implementation, use the JDefinedClass to look up the field for the propertyName.  If the field has the nullable annotation, then add it to the argument to the JMethod object passed to the annotator.

**Comment by recombinationgroup** [comment]
Good idea @ctrimble  💡 I just tried that approach, and it was very straightforward in principle. The extensibility using custom annotators is an excellent feature!

However, I noticed that I then apparently hit https://github.com/joelittlejohn/jsonschema2pojo/issues/1127 , where properties marked as `required` in the schema still get annotated with `@javax.annotation.Nullable`. This means that the underlying JSR305 annotations are faulty, and I can't use them for deriving the getter annotations.

I then noticed that the JSR303 annotations were correctly applied, and tried to derive the remaining annotations from them, using my custom annotator. However, my annotator didn't encounter the JSR303 annotations on its pass over the classes. I then  logged all annotations my custom annotator sees on each field, and noticed that it only sees some of the final annotations on the properties. I was able to see e.g. `javax.annotation.Nullable`, but not `javax.annotation.Nonnull`, and not `javax.validation.constraints.NotNull`. I assume that this is due to an annotator ordering? Is it possible to influence this ordering?
